mono/metadata/method-builder.c

```diff
@@ -663,8 +663,10 @@ mono_mb_emit_exception_for_error (MonoMethodBuilder *mb, MonoError *error)
 	 * If at some point there is need to support other types of errors,
 	 * the behaviour should conform with mono_error_prepare_exception().
 	 */
-	g_assert (mono_error_get_error_code (error) == MONO_ERROR_GENERIC && "Unsupported error code.");
-	mono_mb_emit_exception_full (mb, "System", mono_error_get_exception_name (error), mono_error_get_message (error));
+	g_assert (mono_error_get_exception_name (error) && "Unsupported error code.");
+
+	//FIXME handle exception message in mono_mb_emit_exception_full as it's heap allocated.
+	mono_mb_emit_exception_full (mb, mono_error_get_exception_name_space (error), mono_error_get_exception_name (error), NULL/*, mono_error_get_message (error)*/);
```

Comment:
I don't think I understand the problem - `mono_mb_emit_exception_full` calls `mono_mb_emit_ldstr`.  All the other callers of `mono_mb_emit_ldstr` pass a freshly malloc'd string to it.  Isn't that exactly what we have here?